File: src/main/java/io/golos/commun4J/model/VestingStartRequest.java
package io.golos.commun4J.model;

import com.memtrip.eos.abi.writer.Abi;
import com.memtrip.eos.abi.writer.BytesCompress;
import com.memtrip.eos.abi.writer.NameCompress;

@Abi
public class VestingStartRequest {
    private CommunName owner;

    public VestingStartRequest(CommunName owner) {
        this.owner = owner;
    }

    @NameCompress
    public String getOwner() {
        return owner.getName();
    }

    @BytesCompress
    public byte[] getDecsBytes() {
        return new byte[]{3, 71, 76, 83, 0, 0, 0, 0};
    }

    @NameCompress
    public String getRamPayer() {
        return "eosio";
    }
}

File: src/main/java/io/golos/commun4J/model/VestingStartRequest.kt
package io.golos.commun4J.model

import com.memtrip.eos.abi.writer.Abi
import com.memtrip.eos.abi.writer.BytesCompress
import com.memtrip.eos.abi.writer.NameCompress

@Abi
internal class VestingStartRequest(private val owner: CommunName,
                                   private val ramPayer: CommunName) {

    val getOwner: String
        @NameCompress get() = owner.name

    val decsBytes: ByteArray
        @BytesCompress
        get() = byteArrayOf(3, 71, 76, 83, 0, 0, 0, 0)

    val getRamPayer: String
        @NameCompress
        get() = ramPayer.name


}